Exec/RegTests/EB-ConvergingNozzle/prob.H

```diff
@@ -0,0 +1,126 @@
+#ifndef _PROB_H_
+#define _PROB_H_
+
+#include <AMReX_Print.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_FArrayBox.H>
+#include <AMReX_REAL.H>
+#include <AMReX_GpuMemory.H>
+#include <AMReX_ParmParse.H>
+
+#include "IndexDefines.H"
+#include "Constants.H"
+#include "EOS.H"
+#include "Tagging.H"
+#include "Transport.H"
+#include "ProblemDerive.H"
+#include "prob_parm.H"
+#include "EB.H"
+#include <AMReX_EB2_IF_Rotation.H>
+#include "Utilities.H"
+
+void parse_params(std::unique_ptr<ProbParmDevice>& prob_parm_device);
+
+void EBConvergingNozzle(const amrex::Geometry& geom, const int max_level);
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+void
+pc_initdata(
+  int i,
+  int j,
+  int k,
+  amrex::Array4<amrex::Real> const& state,
+  amrex::GeometryData const& /*geomdata*/,
+  ProbParmDevice const& prob_parm)
+{
+
+  const amrex::Real u = 0.0;
+  const amrex::Real v = 0.0;
+  const amrex::Real w = 0.0;
+  const amrex::Real p = prob_parm.p0;
+  const amrex::Real T = prob_parm.T0;
+  amrex::Real massfrac[NUM_SPECIES] = {1.0};
+  amrex::Real rho, eint;
+  auto eos = pele::physics::PhysicsType::eos();
+  eos.PYT2RE(p, massfrac, T, rho, eint);
+
+  state(i, j, k, URHO) = rho;
+  state(i, j, k, UMX) = rho * u;
+  state(i, j, k, UMY) = rho * v;
+  state(i, j, k, UMZ) = rho * w;
+  state(i, j, k, UEINT) = rho * eint;
+  state(i, j, k, UEDEN) = rho * (eint + 0.5 * (u * u + v * v + w * w));
+  state(i, j, k, UTEMP) = T;
+  for (int n = 0; n < NUM_SPECIES; n++) {
+    state(i, j, k, UFS + n) = rho * massfrac[n];
+  }
+}
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+void
+bcnormal(
```

Comment:
I have a version of this that has the subsonic outlet BC from Blazek implemented here as well (based on EB-C10). With the current FOExtrap BC, the pressure in the nozzle builds over time (it is not defined at either the inlet or outlet), but the Blazek BC holds it constant. However, in the current BC seems to be better at letting pressure waves escape the domain in test runs that I did. The outlet BC is not really relevant for what this test is targetting so it may not matter much how we specify the outlet BC.

Reply:
This is a good point, and might be responsible for weird looking flows I'm seeing in a different configuration.  @emotheau has suggested to fix the pressure at the outflow, extrap T, U and composition, as a simple approach to what it seems like @baperry2 is suggesting.